tests/gtest/avifgridapitest.cc

```diff
@@ -15,139 +14,233 @@ using testing::ValuesIn;
+          cell.width, cell.height, /*depth=*/8, yuv_format, AVIF_PLANES_ALL));
+      if (!cell_images.back()) {
+        return AVIF_RESULT_INVALID_ARGUMENT;
+      }
+      testutil::FillImageGradient(cell_images.back().get());
     }
-    ASSERT_NE(cell_images.back(), nullptr);
-    testutil::FillImageGradient(cell_images.back().get());
   }
 
-  // Encode the grid image.
+  // Encode the grid image (losslessly for easy pixel-by-pixel comparison).
   testutil::AvifEncoderPtr encoder(avifEncoderCreate(), avifEncoderDestroy);
-  ASSERT_NE(encoder, nullptr);
+  if (!encoder) {
+    return AVIF_RESULT_OUT_OF_MEMORY;
+  }
   encoder->speed = AVIF_SPEED_FASTEST;
-  // Just here to match libavif API.
+  encoder->minQuantizer = AVIF_QUANTIZER_LOSSLESS;
+  encoder->maxQuantizer = AVIF_QUANTIZER_LOSSLESS;
+  encoder->minQuantizerAlpha = AVIF_QUANTIZER_LOSSLESS;
+  encoder->minQuantizerAlpha = AVIF_QUANTIZER_LOSSLESS;
+  // cell_image_ptrs exists only to match the libavif API.
   std::vector<avifImage*> cell_image_ptrs(cell_images.size());
   for (size_t i = 0; i < cell_images.size(); ++i) {
     cell_image_ptrs[i] = cell_images[i].get();
   }
-  const avifResult result = avifEncoderAddImageGrid(
-      encoder.get(), horizontal.count, vertical.count, cell_image_ptrs.data(),
+  avifResult result = avifEncoderAddImageGrid(
+      encoder.get(), static_cast<uint32_t>(cell_rows.front().size()),
+      static_cast<uint32_t>(cell_rows.size()), cell_image_ptrs.data(),
       AVIF_ADD_IMAGE_FLAG_SINGLE);
+  if (result != AVIF_RESULT_OK) {
+    return result;
+  }
+
+  testutil::AvifRwData encoded_avif;
+  result = avifEncoderFinish(encoder.get(), &encoded_avif);
+  if (result != AVIF_RESULT_OK) {
+    return result;
+  }
+
+  // Decode the grid image.
+  testutil::AvifImagePtr image(avifImageCreateEmpty(), avifImageDestroy);
+  testutil::AvifDecoderPtr decoder(avifDecoderCreate(), avifDecoderDestroy);
+  if (!image || !decoder) {
+    return AVIF_RESULT_OUT_OF_MEMORY;
+  }
+  result = avifDecoderReadMemory(decoder.get(), image.get(), encoded_avif.data,
+                                 encoded_avif.size);
+  if (result != AVIF_RESULT_OK) {
+    return result;
+  }
 
-  if (expected_success) {
-    ASSERT_EQ(result, AVIF_RESULT_OK);
-    testutil::AvifRwData encoded_avif;
-    ASSERT_EQ(avifEncoderFinish(encoder.get(), &encoded_avif), AVIF_RESULT_OK);
-
-    // Decode the grid image.
-    testutil::AvifImagePtr image(avifImageCreateEmpty(), avifImageDestroy);
-    ASSERT_NE(image, nullptr);
-    testutil::AvifDecoderPtr decoder(avifDecoderCreate(), avifDecoderDestroy);
-    ASSERT_NE(decoder, nullptr);
-    ASSERT_EQ(avifDecoderReadMemory(decoder.get(), image.get(),
-                                    encoded_avif.data, encoded_avif.size),
+  // Reconstruct the input image by merging all cells into a single avifImage.
+  testutil::AvifImagePtr grid =
+      testutil::CreateImage((int)image->width, (int)image->height, /*depth=*/8,
```

Comment:
I consulted the Style Guide. It says:

https://google.github.io/styleguide/cppguide.html#Casting

> - Use brace initialization to convert arithmetic types (e.g., `int64_t{x}`). This is the safest approach because code will not compile if conversion can result in information loss. The syntax is also concise.

Since the example uses a variable `x`, I think `int{}` is not limited to just literals and constants. But it is definitely fine to use `static_cast<int>`.